### PR TITLE
DEV9: Enable pcap non-blocking

### PR DESCRIPTION
**Pull request by TheLastRar**
### Description of Changes
Uses non-blocking mode for pcap

### Rationale behind Changes
When trying to close PCSX2, The Rx networking thread would hang trying to receive a packet
Use non-blocking mode to avoid the hang

### Suggested Testing Steps
Test network enabled games using pcap
Test intercept DHCP or the internal DNS, as non-blocking mode uses a different path (seemed fine in my testing)
Test performance between master and this PR when networking is enabled

### CODE REVIEW

**Comment by F0bes** [comment]
Seems to fix the Waiting on RX thread problem. I don't have any networking home brew set up yet. Could someone test if networking still works on Linux?

**Review by F0bes — APPROVED** [review]
image
Works for me, and the DHCP intercept too.

File: pcsx2/DEV9/pcap_io.cpp [hunk 1]
```diff
@@ -370,6 +379,12 @@ PCAPAdapter::PCAPAdapter()
 	host_mac = hostMAC;
 	ps2_mac = newMAC; //Needed outside of this class
 
+	if (pcap_io_init(config.Eth, config.EthApi == NetApi::PCAP_Switched, newMAC) == -1)
+	{
+		Console.Error("Can't open Device '%s'\n", config.Eth);
```
Comment (lightningterror) [inline pcsx2/DEV9/pcap_io.cpp:384]: DEV9: prefix would be nice for the log, also remove the \n in the end.

Comment (TheLastRar) [reply pcsx2/DEV9/pcap_io.cpp]: Done

**Review by Mrlinkwii — APPROVED** [review]

**Pull request merged**